Content.Server/Psionics/PsionicsSystem.cs

```diff
@@ -304,7 +304,16 @@ public void RerollPsionics(EntityUid uid, PsionicComponent? psionic = null, floa
         psionic.CanReroll = false;
         RollPsionics(uid, psionic, true, bonusMuliplier);
     }
-
+    /// <summary>
+    ///     This function attempts to restore someone's Reroll
+    /// </summary>
+    public void RestorePsionicReroll(EntityUid uid, PsionicComponent? psionic = null)
+    {
+        if (!Resolve(uid, ref psionic, false)
+            || !psionic.Removable)
```

Comment:
```suggestion
            || !psionic.Roller)
```
Removable isn't used anymore for flagging if an entity participates in psionic rolls. So you'd check for !psionic.Roller now.